diff --git a/copasi/CMakeConsoleApp.cmake b/copasi/CMakeConsoleApp.cmake
index f54f49a0f..c63195d07 100644
--- a/copasi/CMakeConsoleApp.cmake
+++ b/copasi/CMakeConsoleApp.cmake
@@ -1,12 +1,12 @@
-# Copyright (C) 2017 by Pedro Mendes, Virginia Tech Intellectual 
-# Properties, Inc., University of Heidelberg, and University of 
-# of Connecticut School of Medicine. 
-# All rights reserved. 
+# Copyright (C) 2017 by Pedro Mendes, Virginia Tech Intellectual
+# Properties, Inc., University of Heidelberg, and University of
+# of Connecticut School of Medicine.
+# All rights reserved.
 
-# Copyright (C) 2012 - 2016 by Pedro Mendes, Virginia Tech Intellectual 
-# Properties, Inc., University of Heidelberg, and The University 
-# of Manchester. 
-# All rights reserved. 
+# Copyright (C) 2012 - 2016 by Pedro Mendes, Virginia Tech Intellectual
+# Properties, Inc., University of Heidelberg, and The University
+# of Manchester.
+# All rights reserved.
 
 # TODO set libraries and library paths
 # for expat, raptor, libsbml
@@ -34,7 +34,7 @@ endif()
 
 if (APPLE)
   FIND_LIBRARY(COREFOUNDATION_LIBRARY CoreFoundation REQUIRED)
+  set(COREFOUNDATION_LIBRARY "/System/Library/Frameworks/CoreFoundation.framework")
   MARK_AS_ADVANCED (COREFOUNDATION_LIBRARY)
   SET(SE_EXTERNAL_LIBS ${SE_EXTERNAL_LIBS} ${COREFOUNDATION_LIBRARY})
 endif (APPLE)
-
diff --git a/copasi/bindings/R/patch-swig-file.cmake b/copasi/bindings/R/patch-swig-file.cmake
index 0d64a38d4..968c0f9fc 100644
--- a/copasi/bindings/R/patch-swig-file.cmake
+++ b/copasi/bindings/R/patch-swig-file.cmake
@@ -1,34 +1,26 @@
-# Copyright (C) 2017 by Pedro Mendes, Virginia Tech Intellectual 
-# Properties, Inc., University of Heidelberg, and University of 
-# of Connecticut School of Medicine. 
-# All rights reserved. 
+# Copyright (C) 2017 by Pedro Mendes, Virginia Tech Intellectual
+# Properties, Inc., University of Heidelberg, and University of
+# of Connecticut School of Medicine.
+# All rights reserved.
 
-# Copyright (C) 2012 - 2016 by Pedro Mendes, Virginia Tech Intellectual 
-# Properties, Inc., University of Heidelberg, and The University 
-# of Manchester. 
-# All rights reserved. 
+# Copyright (C) 2012 - 2016 by Pedro Mendes, Virginia Tech Intellectual
+# Properties, Inc., University of Heidelberg, and The University
+# of Manchester.
+# All rights reserved.
 
 if (NOT FILENAME OR NOT EXISTS ${FILENAME})
   message(FATAL_ERROR "Please specify the filename to the file to patch")
 endif()
 
-if (EXISTS "${SRC_DIR}/sedscript.txt")
-execute_process(COMMAND sed 
-          -f ${SRC_DIR}/sedscript.txt -i
-           "${FILENAME}"
-           )
-endif()
-
 file(READ "${FILENAME}" SOURCECODE)
 
+# Needed on Ubuntu swig 3.0.10
+# Not needed on Mac swig 3.0.12
 string(REPLACE "'get''get'" "'get','get'" SOURCECODE "${SOURCECODE}" )
-string(REPLACE "'get''get'" "'get','get'" SOURCECODE "${SOURCECODE}" )
-string(REPLACE "'get''get'" "'get','get'" SOURCECODE "${SOURCECODE}" )
-string(REPLACE ", \"}" ", \"" SOURCECODE "${SOURCECODE}" )
-string(REPLACE "\${enumTo" "enumTo" SOURCECODE "${SOURCECODE}" )
-string(REPLACE "\${enumFrom" "enumFrom" SOURCECODE "${SOURCECODE}" )
-string(REPLACE "if ( &&" "if (" SOURCECODE "${SOURCECODE}" )
-string(REPLACE "if ()" "if(TRUE)" SOURCECODE "${SOURCECODE}" )
+
+# Needed on Ubuntu swig 3.0.10
+# Needed on Mac swig 3.0.12
 string(REPLACE "ans <- new(\"_p_size_t\", ref=ans) ;" "" SOURCECODE "${SOURCECODE}" )
+
 file(WRITE "${FILENAME}" "${SOURCECODE}")
 message (STATUS "Patched COPASI.R")
